### PR TITLE
added upload and download aliases in csv format.

### LINKED ISSUE #1793: @aliases endpoint should allow to upload/download all aliases and allow to download filtered aliases as CSV file
PR for adding aliases: https://github.com/plone/plone.restapi/pull/1398

The @aliases should allow to upload aliases as CSV files and to download all existing aliases as CSV file. I am guessing there is a browser view in ClassicUI that does this already. Therefore it should be pretty straight forward to add this to the existing controlpanel. 

However, it would be nice if we would allow to download a CSV file based on any filter criterion of the @aliases endpoint:

http://localhost:8080/Plone/++api++/@aliases&q=/de/ft/my-path&manual=true&from=2021-01-01&to=2023-01-01

We could use content negotiation for this and send the "Accept" HTTP header "text/csv". Then REST API would return the CSV file.

# Code

Download method of the redirects control panel:

https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/controlpanel/browser/redirects.py#L457

There is not a single unit/integration test for the aliases service. :(

### PR DESCRIPTION
**Pull request by Faakhir30**
Refers #1812 

Closes #1793 

Using content negotiation for PUT, GET aliases service.
 - send the "Accept" HTTP header "text/csv" for get request to download aliases as csv. Then REST API would return the CSV file.
 - send the "Content-Type" HTTP header "text/csv" for post request to add aliases from csv.

Tasks breakdown:

- [x] Handle `text/csv` content-type to upload aliases in bulk.
- [x] Handle `text/csv` content-type to download aliases.
- [x] Add new http-examples
- [x] Add aliases unit/integration tests to `plone.restapi`.
- [x] Add/update documentation

----
📚 Documentation preview 📚: https://plonerestapi--1813.org.readthedocs.build/

### CODE REVIEW

**Comment by Faakhir30** [comment]
@jenkins-plone-org please run jobs

**Comment by Faakhir30** [comment]
@jenkins-plone-org please run jobs

**Comment by stevepiercy** [comment]
RTD preview looks good.

https://plonerestapi--1813.org.readthedocs.build/en/1813/endpoints/aliases.html#adding-url-aliases-in-bulk-via-csv

This is ready for technical review.

I think the Jenkins failures are flaky tests. I started a retry for the two failures.

**Review by Petchesi-Iulian — APPROVED** [review]
LGTM codewise, looks like some test is failing

**Comment by Faakhir30** [comment]
The test failing `plone.app.contenttypes.tests.test_robot.RobotTestCase.Scenario Test Relative Location Criterion`
 still looks unstable/flaky to me, read from plone/plone.app.contenttypes#362.

Also, I'm unable to understand its relation with this `aliases service`.
Maybe someone who has experience with `Robot Tests` should chime in. @stevepiercy

**Comment by stevepiercy** [comment]
Previously two Jenkins builds failed, now it's down to one. I started one more retry.

https://jenkins.plone.org/job/pull-request-6.1-3.12/639/

Those failures are totally unrelated to this PR, so I'm OK with ignoring them.

File: src/plone/restapi/services/aliases/add.py [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
+from zope.component.hooks import getSite
 from zope.component import getUtility
 from zope.interface import alsoProvides
 from zope.interface import implementer
 from zope.publisher.interfaces import IPublishTraverse
 
 import plone.protect.interfaces
+import logging
+
+logger = logging.getLogger("Plone")
```
Comment (davisagli) [inline src/plone/restapi/services/aliases/add.py:21]: Please change to `logging.getLogger(__name__)` to match what we do elsewhere.

File: src/plone/restapi/services/aliases/add.py [hunk 2]
```diff
@@ -83,15 +90,35 @@ def edit_for_navigation_root(self, alias):
 class AliasesRootPost(Service):
     """Creates new aliases via controlpanel"""
 
-    def reply(self):
-        data = json_body(self.request)
+    def _reply_csv(self):
+        form = self.request.form
+        if not form.get("file"):
+            raise BadRequest("No file uploaded")
```
Comment (davisagli) [inline src/plone/restapi/services/aliases/add.py:96]: We should also check that the uploaded file has content-type text/csv

**Review by davisagli — COMMENTED** [review]
@Faakhir30 In general I'd like to have a chance to look at new features before they are merged, and I hadn't had a chance to look at this one yet. But it looks like good work. Thank you!

**Comment by Faakhir30** [comment]
@davisagli understood, will make sure of your review on feature PRs from now on :)
Will create a patch PR for these changes Requested

File: src/plone/restapi/services/aliases/add.py (hunk 2 above)
Comment (Faakhir30) [reply src/plone/restapi/services/aliases/add.py:96]: @davisagli  I dont think so this is possilbe ive tried locally, In `Zope`, the `ZPublisher.HTTPRequest.FileUpload` object does not directly expose a `content_type` attribute, So this file is not having any `content_type` attribute.

Comment (Faakhir30) [reply src/plone/restapi/services/aliases/add.py:96]: Fixed at #1821

**Pull request merged**